Added Gap analysis from Google Sheet. This is the v03 for CCAMP WG adoption

reza-github opened the pull request:
Added Gap analysis from Google Sheet. This is the v03 for CCAMP WG adoption

reza-github committed:
add Gap analysis from Google Sheet

add Gap analysis from Google Sheet

reza-github committed:
added more references from OC, TAPI, ITU-T

added more references from OC, TAPI, ITU-T

reza-github committed:
Added more reference for OC, ITU-T and TAPI

Added more reference for OC, ITU-T and TAPI

reza-github committed:
Enhanced sections 6, 7

Enhanced sections 6, 7

aguoietf reviewed (commented):
Also add reference to draft-ietf-ccamp-wdm-tunnel-yang and draft-ietf-ccamp-optical-path-computation-yang which should be part of the gap analysis for controller NBI

manzoro reviewed (commented):
I've not reviewed for the moment the list and description of attribute gaps taken from google sheet, we have discussed for some time around them and I agree we can pull them in the current status for -03 revision and adoption.
We may probably need a second round of review and re-organization of the same  as a next step.

ggrammel reviewed (approved):
Comments mostly of editorial nature and softening a few rough edges.

reza-github committed:
added more IETF references

added more IETF references

reza-github committed:
Added more IETF references

Added more IETF references

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
-(Q: how can we incorporate the Google Sheet?).
+* IETF YANG  data model to manage configurable DWDM optical interfaces {{?I-D.ietf-ccamp-optical-impairment-topology-yang}}
+* IETF YANG Data Model for Optical Impairment-aware Topology {{?I-D.ietf-ccamp-rfc9093-bis}}
+* IETF Common YANG Data Types for Layer 0 Networks {{?I-D.ietf-ccamp-dwdm-if-param-yang}}
```
This is the order in version03.

This can be CLOSED

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+
+(Q: how can we incorporate the Google Sheet? Can we use download? 
+
+You can [download the Google sheet](/Users/rrokui/Desktop/optical-pluggable-attributes-v00-GAP.xslx) to see all the details.
```
I will add the Editorial note except the last sentence. However, why we should remove to Google sheet? When we are done with gap analysis, the google sheet should be ported to IETF Github or somewhere else to make sure everyone can access it.

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+| 198       | degradation in signal quality, measured as an increase |
+|           | in required OSNR to achieve a target BER               |
+| --------- | ------------------------------------------------------ |
+
```
IMO the table format is better option as in your example it is not clear what for example [4] means or what (IETF) means?
Roberto having said thatRoberto, let's keep the existing format for V03 version. For future version of this document we can discuss this topic.

ggrammel reviewed (approved):
approved when suggestions included

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+Note that the "Attribute Number" refers to numbering in Google Sheet.
 
-The following items are identified as initial gap related to optical modules. Note that the complete list will be provided after finishing the Google Sheet.
+Incorporating these missing capabilities into IETF YANG models would significantly enhance the modeling of optical pluggable modules by bridging the gap between protocol-level management and physical layer specifications. It would facilitate unified management across heterogeneous network elements supporting diverse optical standards and proprietary extensions, thereby improving automation, opera ...
```
Your suggestion is incorporated.

This can be CLOSED

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+## List of Missing Optical Pluggable Configuration Attributes Based on Gap Analysis
+
+
+{{figure-gap-analysis-Configuration}} highlights missing configuration attributes for optical pluggables. These parameters and features, essential for proper operation and performance, are often defined by other standards development organizations (SDOs) and industry forums but may not be fully represented in IETF YANG models. Integrating these missing configuration attributes into IETF YANG mode ...
```
Your suggestion is incorporated.

This can be CLOSED

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+
+As discussed in {{plug-capabilities-attributes}}, {{plug-config-attribute}}, and {{plug-pm-definition}}, the google sheet provides the Capabilities, Configuration, and Performance Monitoring attributes for coherent pluggable. The google sheet includes various read-only capability attributes, read-write configuration attributes, and read-only performance monitoring attributes. For a comprehensive  ...
+
+(Q: how can we incorporate the Google Sheet? Can we use download? 
```
Does IETF have any GitHub or SharePoints?

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+
+Addressing syntax gaps in coherent pluggable attributes is crucial for achieving consistent and interoperable management across different implementations. One key issue lies in establishing a standardized naming convention. There are some naming inconsistency on existing IETF drafts {{?I-D.ietf-ccamp-optical-impairment-topology-yang}}, {{?I-D.ietf-ccamp-rfc9093-bis}} and {{?I-D.ietf-ccamp-dwdm-if ...
+
+{{figure-gap-syntax}} shows a proposal for naming conventiion of optical pluggacle attributes.  The current proposal, while illustrative, may need further refinement to ensure clarity and avoid ambiguity. Specifically, the prefixing of attributes with directions like "tx" or "rx" should be consistently applied to all relevant parameters to clearly identify the directionality of the signal. The ex ...
```
Corrected and done.

This can be CLOSED

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+| --------- | ------------------------------------------ | --------- |
+| 4         | tag-id                                     | IETF      |
+| 16        | Post FEC BER                               | oif       |
+| 17        | pre-fec-ber                                |           |
```
Sergio any details of the tables will be addressed after issuing the V03. After submitting V03 of document (for WC adoption), we start these tables and Gap analysis right away.

This can be discussed after V03 (for WC adoption).

sergiobelotti replied:
Not sure what you're saying is the correct procedure. Submitting a version containing something that is not approved is incorrect unless all the gap analysis part has to be considered "not in review for WG LC". Not sure that can be acceptable by chairs.

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+| 4         | tag-id                                     | IETF      |
+| 16        | Post FEC BER                               | oif       |
+| 17        | pre-fec-ber                                |           |
+| 18        | Target reach                               | oif/itu-t |
```
Sergio any details of the tables will be addressed after issuing the V03. After submitting V03 of document (for WC adoption), we start these tables and Gap analysis right away.

This can be discussed after V03 (for WC adoption).

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+| 16        | Post FEC BER                               | oif       |
+| 17        | pre-fec-ber                                |           |
+| 18        | Target reach                               | oif/itu-t |
+| 19        | Ripple                                     | oif/itu-t |
```
Sergio any details of the tables will be addressed after issuing the V03. After submitting V03 of document (for WC adoption), we start these tables and Gap analysis right away.

This can be discussed after V03 (for WC adoption).

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,470 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+| 36        | Polarization Rotation Speed                | oif/itu-t |
+| 39        | min-tx-osnr                                | OpenConfg |
+| 46        | pulse-shaping-type                         | OpenConfg |
+| 53        | fec-coding                                 | OpenConfg |
```
Sergio any details of the tables will be addressed after issuing the V03. After submitting V03 of document (for WC adoption), we start these tables and Gap analysis right away.

This can be discussed after V03 (for WC adoption).

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,473 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+|           | nominal central frequency                  |           |
+| 123       | Maximum skew between the two polarizations | itu-t     |
+| 125       | Maximum spectral power density             | itu-t     |
+| 126       | Maximum TDECQ                              | itu-t     |
```
Sergio any details of the tables will be addressed after issuing the V03. After submitting V03 of document (for WC adoption), we start these tables and Gap analysis right away.

This can be discussed after V03 (for WC adoption).

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,473 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+| 125       | Maximum spectral power density             | itu-t     |
+| 126       | Maximum TDECQ                              | itu-t     |
+| 127       | Maximum I-Q offset                         | itu-t     |
+| 157       | Laser frequency accuracy                   | oif       |
```
Sergio any details of the tables will be addressed after issuing the V03. After submitting V03 of document (for WC adoption), we start these tables and Gap analysis right away.

This can be discussed after V03 (for WC adoption).

reza-github commented on draft-rokui-ccamp-actn-wdm-pluggable-modelling.md:
```diff
@@ -545,33 +585,473 @@ Note that sometimes these thresholds are configurable and sometime they are hard
+| 126       | Maximum TDECQ                              | itu-t     |
+| 127       | Maximum I-Q offset                         | itu-t     |
+| 157       | Laser frequency accuracy                   | oif       |
+| 158       | Laser frequency noise                      | oif       |
```
Sergio any details of the tables will be addressed after issuing the V03. After submitting V03 of document (for WC adoption), we start these tables and Gap analysis right away.

This can be discussed after V03 (for WC adoption).

reza-github committed:
Incorporated suggestions from PR#18

Incorporated suggestions from PR#18 (Added Gap analysis from Google Sheet. This is the v03 for CCAMP WG adoption)

reza-github committed:
Added Editorial notes for Section 7

Added Editorial notes for Section 7 provided by Sergio and Julian
